torch/csrc/distributed/c10d/ProcessGroupNCCL.cpp

```diff
@@ -1227,7 +1227,21 @@ std::string ProcessGroupNCCL::getNCCLWatchdogDebugInfo() {
   return retrieveDesyncReport(store_, "NCCL", rank_, size_);
 }
 
-void ProcessGroupNCCL::workCleanupLoop() {
+bool ProcessGroupNCCL::shouldAutoDumpWorkTimeOut(WorkNCCL& work) {
+  if (!dumpOnTimeout_)
+    return false;
+  auto timeout = std::chrono::milliseconds(heartbeatTimeoutInSec_ * 1000);
+  return work.checkTimeout(c10::optional<std::chrono::milliseconds>(timeout));
+}
+
+void ProcessGroupNCCL::checkAndSetStore() {
+  if (!store_->check({std::string(TIMEOUT_DUMP)})) {
```

Comment:
Is there a need to check TIMEOUT_DUMP here vs just setting it? It is 1/2 the messages to just set and would be a race anyway.

Reply:
Right, we can just set, because we don't care about race condition here.